### PR TITLE
nixos/metricbeat: init

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

Add a module for metricbeat, the metric shipper for the ELK stack.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117379` at 32649f76 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>logstash (logstash6)</li>
    <li>logstash6-oss</li>
    <li>logstash7</li>
    <li>logstash7-oss</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/misc/logstash/6.x.nix:29:3:
```
   |
29 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/misc/logstash/7.x.nix:29:3:
```
   |
29 |   buildInputs = [
   |   ^
```
</li>
  </ul>
</details>

File: nixos/tests/elk.nix [hunk 1]
```diff
@@ -203,11 +241,13 @@ in pkgs.lib.mapAttrs mkElkTest {
...
+      metricbeat    = pkgs.metricbeat7;
     }
     else {
       elasticsearch = pkgs.elasticsearch7-oss;
       logstash      = pkgs.logstash7-oss;
       kibana        = pkgs.kibana7-oss;
       journalbeat   = pkgs.journalbeat7;
-    };
+      metricbeat    = pkgs.metricbeat7;
+  };
```
Comment (zseri) [inline nixos/tests/elk.nix:252]: The indention of this line seems to be a bit off...

Comment (roberth) [reply nixos/tests/elk.nix]: Fixed. Thanks.

**Comment by roberth** [comment]
@ofborg build metricbeat

**Comment by roberth** [comment]
cc elasticsearch maintainers @apeschar @basvandijk. Could you please review this "ES-ecosystem" pr?

File: nixos/modules/services/monitoring/metricbeat.nix [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    systemd.services.metricbeat = {
+      description = "metricbeat metrics shipper";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = ''
+          ${cfg.package}/bin/metricbeat \
+            -c ${settingsFormat.generate "metricbeat.yml" cfg.settings} \
+            --path.data $STATE_DIRECTORY \
+            --path.logs $LOGS_DIRECTORY \
+            ;
```
Comment (zseri) [inline nixos/modules/services/monitoring/metricbeat.nix:141]: what's the semicolon needed for?

Comment (roberth) [reply nixos/modules/services/monitoring/metricbeat.nix:141]: It's for the backslash. This style of multi-line statement produces smaller diffs, making it easier to merge.

**Comment by roberth** [comment]
I was hoping for more feedback or an approval, but I've already called attention to this PR a number of times. It's time to move this forward. If anyone has improvements, I can implement them after the merge as well.

Thank you @zseri for your comments.

**Pull request merged**